src/ruby.rs

```diff
@@ -51,100 +51,890 @@ extern {
     pub static rb_eEncodingError: VALUE;
+
+    /// The `Encoding::CompatibilityError` exception class
+    ///
+    /// # Defined In
+    ///
+    /// * **2.5:**
+    ///     [ruby.h](https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L1973) |
+    ///     [documentation](https://ruby-doc.org/core-2.5.1/Encoding/CompatibilityError.html)
     pub static rb_eEncCompatError: VALUE;
 
+    /// The `ScriptError` exception class
+    ///
+    /// # Defined In
+    ///
+    /// * **2.5:**
+    ///     [ruby.h](https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L1975) |
+    ///     [documentation](https://ruby-doc.org/core-2.5.1/ScriptError.html)
     pub static rb_eScriptError: VALUE;
+
+    /// The `NameError` exception class
+    ///
+    /// # Defined In
+    ///
+    /// * **2.5:**
+    ///     [ruby.h](https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L1976) |
+    ///     [documentation](https://ruby-doc.org/core-2.5.1/NameError.html)
     pub static rb_eNameError: VALUE;
+
+    /// The `SyntaxError` exception class
+    ///
+    /// # Defined In
+    ///
+    /// * **2.5:**
+    ///     [ruby.h](https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L1977) |
+    ///     [documentation](https://ruby-doc.org/core-2.5.1/SyntaxError.html)
     pub static rb_eSyntaxError: VALUE;
+
+    /// The `LoadError` exception class
+    ///
+    /// # Defined In
+    ///
+    /// * **2.5:**
+    ///     [ruby.h](https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L1978) |
+    ///     [documentation](https://ruby-doc.org/core-2.5.1/LoadError.html)
     pub static rb_eLoadError: VALUE;
 
+    /// The `Math::DomainError` exception class
+    ///
+    /// # Defined In
+    ///
+    /// * **2.5:**
+    ///     [ruby.h](https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L1980) |
+    ///     [documentation](https://ruby-doc.org/core-2.5.1/Math/DomainError.html)
     pub static rb_eMathDomainError: VALUE;
 
+
+    /// Convert a C string to an [`ID`](struct.id.html)
+    ///
+    /// * `cstr` - nul-terminated C string, treated as ASCII encoded
+    ///
+    /// # Defined In
+    ///
+    /// * **2.5:**
+    ///     [ruby.h](https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L1728) |
+    ///     [symbol.c](https://github.com/ruby/ruby/blob/v2_5_1/symbol.c#L610-L614) |
+    ///     [documentation](https://ruby-doc.org/core-2.5.1/doc/extension_rdoc.html#label-ID+or+Symbol)
     pub fn rb_intern(cstr: *const c_char) -> ID;
-    pub fn rb_intern2(ptr: *const c_char, len: c_long) -> ID;
+
+    /// Convert a C string of the given length to an [`ID`](struct.id.html)
+    ///
+    /// * `cstr` - C string, treated as ASCII encoded
+    /// * `len` - number of C chars
+    ///
+    /// # Miscellaneous
+    ///
+    /// [`rb_intern_const`](https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L1757-L1760)
+    /// is more or less an alias of this.
```

Comment:
While `rb_intern_const` does call this, I don't think "more or less an alias of this" is accurate or helpful in this case.